File: Examen1/2026_1/Plantillas/TestSoft/TestSoftNegocio/src/main/java/pe/edu/pucp/testsoft/negocio/examen/boimpl/ExamenBOImpl.java
package pe.edu.pucp.testsoft.negocio.examen.boimpl;

import pe.edu.pucp.testsoft.model.alumno.Alumno;
import pe.edu.pucp.testsoft.model.examen.Examen;
import pe.edu.pucp.testsoft.model.pregunta.Pregunta;
import pe.edu.pucp.testsoft.negocio.examen.bo.ExamenBO;
import pe.edu.pucp.testsoft.persistance.examen.dao.ExamenDAO;
import pe.edu.pucp.testsoft.persistance.examen.impl.ExamenImpl;

import java.util.List;

public class ExamenBOImpl implements ExamenBO {

    private ExamenDAO examenDAO = new ExamenImpl();

    @Override
    public int insertar(Examen objeto) throws Exception {
        return examenDAO.insertar(objeto);
    }

    @Override
    public int modificar(Examen objeto) throws Exception {
        return examenDAO.modificar(objeto);
    }

    @Override
    public int eliminar(int id) throws Exception {
        return examenDAO.eliminar(id);
    }

    @Override
    public Examen buscarPorId(int id) throws Exception {
        return examenDAO.buscarPorId(id);
    }

    @Override
    public List<Examen> leerTodos() {
        return examenDAO.listarTodos();
    }

    @Override
    public void crearExamenConPreguntas(Alumno alumno, String titulo, List<Pregunta> preguntas) {
        Examen examen = new Examen(alumno, titulo, preguntas);
        int idExamen = examenDAO.insertar(examen);
        System.out.println("Examen creado: " + idExamen + " - Fecha: " + examen.getFechaCreacion());
    }
}

File: Examen1/2026_1/Plantillas/PlantillaC#/TestSoft/TestSoftBusiness/Examenes/BO/ExamenBOImpl.cs
using TestSoftModel.Alumno;
using TestSoftModel.Examen;
using TestSoftModel.Pregunta;
using TestSoftBusiness.Examenes.BOI;
using TestSoftPersistance.Examenes.DAO;
using TestSoftPersistance.Examenes.Impl;

namespace TestSoftBusiness.Examenes.BO
{
    public class ExamenBOImpl : IExamenBO
    {
        private readonly ExamenDAO examenDAO = new ExamenImpl();

        public int Insertar(Examen objeto) => examenDAO.Insertar(objeto);
        public int Modificar(Examen objeto) => examenDAO.Modificar(objeto);
        public int Eliminar(int id) => examenDAO.Eliminar(id);
        public Examen BuscarPorId(int id) => examenDAO.BuscarPorId(id);
        public List<Examen> ListarTodos() => examenDAO.ListarTodos();

        // Crea un examen con sus preguntas (transaccion en la capa DAO)
        public void CrearExamenConPreguntas(Alumno alumno, string titulo, List<Pregunta> preguntas)
        {
            Examen examen = new Examen();
            examen.IdAlumno = alumno.Id;
            examen.Titulo = titulo;
            examen.Preguntas = preguntas;

            int idExamen = examenDAO.Insertar(examen);
            Console.WriteLine($"Examen creado: {idExamen} para {alumno.Codigo}");
        }
    }
}
